### PR TITLE
Break client table cycle

### PR DESCRIPTION
**Pull request by coryan**
This is part of the fixes for #101.  Please read the issue for a
full description of where these changes are headed, in this PR
we "simply" break a dependency cycle between two classes.

### CODE REVIEW

File: bigtable/client/testing/table_test_fixture.h [hunk 1]
```diff
@@ -39,12 +39,22 @@ class TableTestFixture : public ::testing::Test {
...
-  std::string instance_id_ = "the-instance";
+  std::string const kProjectId = "foo-project";
+  std::string const kInstanceId = "bar-instance";
+  std::string const kTableId = "baz-table";
+
+  // This is hardcoded and not computed because we want to test the compuation.
+  std::string const kInstanceName =
+      "projects/foo-project/instances/bar-instance";
+  std::string const kTableName =
+          "projects/foo-project/instances/bar-instance/tables/baz-table";
```
Comment (carterpage) [inline bigtable/client/testing/table_test_fixture.h]: Fix indent.

File: bigtable/client/data.h [hunk 2]
```diff
@@ -45,7 +45,7 @@ std::shared_ptr<ClientInterface> CreateDefaultClient(std::string project_id,
                                                      ClientOptions options);
 
 inline std::string CreateInstanceName(std::shared_ptr<ClientInterface> client) {
```
Comment (carterpage) [inline bigtable/client/data.h]: Why isn't this called GetInstanceName (and the other one GetTableName)?

I generally think of "Create" as an operation that involves changing some state, so was initially confused.

Comment (coryan) [reply bigtable/client/data.h]: Fixed.

It creates a new object (the string).  But never mind, if it is confusing it should be changed.  Also planning to refactor these things in #144.

File: bigtable/client/testing/table_test_fixture.h (hunk 1 above)
Comment (coryan) [reply bigtable/client/testing/table_test_fixture.h]: Fixed.

**Pull request merged**